### PR TITLE
Add support for custom duration in `switchTo().frame()`

### PR DESCRIPTION
**Pull request by donesvad**
## Proposed changes
Add support for custom duration in `switchTo().frame()`

## Checklist
- [x] Checkstyle and unit tests are passed locally with my changes by running `gradlew check chrome_headless firefox_headless` command
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/main/java/com/codeborne/selenide/SelenideTargetLocator.java [hunk 1]
```diff
@@ -41,12 +41,57 @@ public SelenideTargetLocator(Driver driver) {
...
+
+  /**
+   * Switch to frame by index with a configurable timeout
+   * NB! Order of frames can be different in different browsers, see Selenide tests.
+   *
+   * @param index    index of frame (0-based)
+   * @param duration the timeout duration. It overrides default Config.timeout()
+   */
+  @Nonnull
+  public WebDriver frame(int index, Duration duration) {
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/SelenideTargetLocator.java:64]: rename parameter `duration` to `timeout`.

File: src/main/java/com/codeborne/selenide/SelenideTargetLocator.java [hunk 2]
```diff
@@ -41,12 +41,57 @@ public SelenideTargetLocator(Driver driver) {
...
+  }
+
+  /**
+   * Switch to frame by name or id with a configurable timeout
+   *
+   * @param nameOrId name or id of frame
+   * @param duration            the timeout duration. It overrides default Config.timeout()
+   */
+  @Nonnull
+  public WebDriver frame(String nameOrId, Duration duration) {
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/SelenideTargetLocator.java:86]: rename `duration` -> `timeout`

File: src/test/java/integration/FramesCustomWaitTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  @Test
+  void waitsUntilFrameAppears_withLowerTimeout() {
+    $("#btn").click();
+    assertThatThrownBy(() -> switchTo().frame(1, Duration.ofSeconds(1)))
+      .isInstanceOf(FrameNotFoundException.class);
+  }
+
+  @AfterEach
+  void tearDown() {
+    driver().close();
```
Comment (asolntsev) [inline src/test/java/integration/FramesCustomWaitTest.java:42]: we don't need to close the browser after every test: it makes tests much slower.

**Review by asolntsev — CHANGES_REQUESTED** [review]

File: src/main/java/com/codeborne/selenide/SelenideTargetLocator.java (hunk 1 above)
Comment (donesvad) [reply src/main/java/com/codeborne/selenide/SelenideTargetLocator.java]: renamed, thx

File: src/main/java/com/codeborne/selenide/SelenideTargetLocator.java (hunk 2 above)
Comment (donesvad) [reply src/main/java/com/codeborne/selenide/SelenideTargetLocator.java]: renamed, thx

File: src/test/java/integration/FramesCustomWaitTest.java (hunk 3 above)
Comment (donesvad) [reply src/test/java/integration/FramesCustomWaitTest.java]: removed unnecessary closing

**Review by alexandrter — APPROVED** [review]

**Review by asolntsev — APPROVED** [review]

**Pull request merged**